sql/core/src/test/scala/org/apache/spark/sql/DatasetPrimitiveSuite.scala

```diff
@@ -28,6 +28,8 @@ case class SeqClass(s: Seq[Int])
 
 case class ListClass(l: List[Int])
 
+case class JListClass(l: java.util.List[Int])
```

Comment:
This is not supported intentionally. If we want to support java classes and scala classes mixedly, we should unify the `ScalaRelfection` and `JavaTypeInference`, which is a lot of effort.

For now I think it's more important to completely support specific collections type in scala, and then try to unify the `ScalaRelfection` and `JavaTypeInference`.

what do you think?